devicetypes/smartthings/zigbee-window-shade.src/zigbee-window-shade.groovy

```diff
@@ -94,37 +101,60 @@ def parse(String description) {
                     sendEvent(name: "level", value: zigbee.convertHexToInt(liftmap.value))
                 }
             }
+        } else if (!usesWindowCovering() && descMap?.clusterInt == zigbee.LEVEL_CONTROL_CLUSTER && descMap.value) {
+            def valueInt = Math.round((zigbee.convertHexToInt(descMap.value)) / 255 * 100)
+            state.level = valueInt
+            if (0 == valueInt) {
+                sendEvent(name: "windowShade", value: "closed")
+            } else if (100 == valueInt) {
+                sendEvent(name: "windowShade", value: "open")
+            } else {
+                sendEvent(name: "windowShade", value: "partially open")
+            }
+            sendEvent([name: "level", value: valueInt])
         }
     }
 }
 
+def usesWindowCovering() {
+    device.getDataValue("manufacturer") != "Feibit Co.Ltd"
+}
+
 def close() {
     log.info "close()"
     sendEvent(name: "windowShade", value: "closing")
-    zigbee.command(CLUSTER_WINDOW_COVERING, 0x01)
+    zigbee.command(CLUSTER_WINDOW_COVERING, COMMAND_CLOSE)
 }
 
 def open() {
     log.info "open()"
     sendEvent(name: "windowShade", value: "opening")
-    zigbee.command(CLUSTER_WINDOW_COVERING, 0x00)
+    zigbee.command(CLUSTER_WINDOW_COVERING, COMMAND_OPEN)
 }
 
-def setLevel(data) {
+def setLevel(data, rate = null) {
     log.info "setLevel()"
     Integer currentLevel = state.level
     Integer level = data as Integer
+
     if (level > currentLevel) {
         sendEvent(name: "windowShade", value: "opening")
     } else if (level < currentLevel) {
         sendEvent(name: "windowShade", value: "closing")
     }
-    zigbee.command(CLUSTER_WINDOW_COVERING, 0x05, zigbee.convertToHexString(data, 2))
+    def cmd
+    if (usesWindowCovering()) {
+        cmd = zigbee.command(CLUSTER_WINDOW_COVERING, COMMAND_GOTO_LIFT_PERCENTAGE, zigbee.convertToHexString(data, 2))
+    } else {
+        cmd = zigbee.command(zigbee.LEVEL_CONTROL_CLUSTER, COMMAND_MOVE_LEVEL_ONOFF, zigbee.convertToHexString(Math.round(data * 255 / 100), 2))
```

Comment:
So is it the case that the Wistar supports the Window Covering Cluster but not the GOTO_LIFT_PERCENTAGE command or the LIFT_PERCENTAGE attribute? If that's the case, perhaps it would be more clear to change `usesWindowCovering` to `supportsLiftPercentage`

Reply:
No, I have try the GOTO_LIFT_PERCENTAGE command and the LIFT_PERCENTAGE attribute, the device return useless information. I will update this.